python/ray/train/v2/_internal/execution/train_fn_utils.py

```diff
@@ -0,0 +1,90 @@
+import threading
+from typing import Any, Dict, Optional
+
+from ray.data import DataIterator
+from ray.train import Checkpoint
+from ray.train.v2._internal.execution.context import (
+    get_train_context as get_internal_train_context,
+)
+from ray.train.v2.api.context import TrainContext as PublicTrainContext
+
+
+class TrainFnUtils:
+    """Utility class providing an abstraction layer between user-facing APIs
+        and :class:`~ray.train.v2._internal.execution.context.TrainContext`.
+
+    This is one layer between user facing APIs and :class:`~ray.train.v2._internal.execution.context.TrainContext`.
```

Comment:
nit: remove this sentence since you said the same thing in the previous sentence